catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryList.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.graphics.PointF;
+
+import java.util.ArrayList;
+
+public class EmbroideryList extends ArrayList{
+
+	@Override
+	public synchronized boolean add(Object o) {
+		boolean status = false;
+		PointF point = (PointF) o;
+
+		if (super.isEmpty() || isNewStitchPoint(point)) {
+			status = super.add(point);
+		}
+
+		return status;
+	}
+
+	private Boolean isNewStitchPoint(PointF point) {
+		return !super.get(super.size() - 1).equals(point);
```

Comment:
This only checks the last point in the list. What if the point is somewhere else in the list (and therefore not new)? Could this happen?

Reply:
i know ticket does not reflect what was done here. (meetings/mails with bernadette)

When a stitchpoint is added it is important that it is not exactly the same point again, but it is fine to have the same point e.g. three stitches later again. This prevents that a user stitches again and again and again on the same position which will destory the needle at some point. In the future there will also be a check introduced that the same point can't be more often in that list than some threshold x.